airbyte-integrations/connectors/source-zendesk-support/unit_tests/unit_test.py

```diff
@@ -21,7 +21,113 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 #
+import json
+import pendulum
+from unittest import TestCase
+import timeout_decorator
+import requests_mock
+from airbyte_cdk.sources.streams.http.exceptions import UserDefinedBackoffException
 
-#
-def test_dummy_test():
-    pass
+from source_zendesk_support import SourceZendeskSupport
+from source_zendesk_support.streams import (
+    Tickets,
+    Users,
+    TicketMetrics,
+    Macros,
+    TicketAudits,
+    Tags
+)
+
+CONFIG_FILE = "secrets/config.json"
+
+
+class TestZendeskSupport(TestCase):
+    """ This test class provides a set of tests for different Zendesk streams.
+          The Zendesk API has difference pagination and sorting mechanisms for streams.
+          Let's try to check them
+        """
+    @staticmethod
+    def prepare_stream_args():
+        """Generates streams settings from a file"""
+        with open(CONFIG_FILE, "r") as f:
+            return SourceZendeskSupport.convert_config2stream_args(json.loads(f.read()))
+
+    def _test_export_stream(self, stream_cls: type):
+        stream = stream_cls(**self.prepare_stream_args())
+        record_timestamps = {}
+        for record in stream.read_records(sync_mode=None):
+            # save the first 5 records
+            if len(record_timestamps) > 5:
+                break
+            record_timestamps[record['id']] = record[stream.cursor_field]
+        for record_id, timestamp in record_timestamps.items():
```

Comment:
stopping the sync prematurely is clever; good idea. We should also verify the cursor field is `>=` input cursor field from the state object. 

We should probably incorporate this into SAT, something similar to #4985 but leveraged in the SAT.